sktime/forecasting/compose/_skforecast_reduce.py

```diff
@@ -0,0 +1,445 @@
+        "capability:pred_int": True,
+        "capability:pred_int:insample": False,
+        "python_version": ">=3.8,<3.13",
+        "python_dependencies": ["skforecast>=0.12.1"],
+    }
+
+    def __init__(
+        self: "SkforecastAutoreg",
+        regressor: object,
+        lags: Union[int, np.ndarray, list],
+        transformer_y: Optional[object] = None,
+        transformer_exog: Optional[object] = None,
+        weight_func: Optional[Callable] = None,
+        differentiation: Optional[int] = None,
+        fit_kwargs: Optional[dict] = None,
+        binner_kwargs: Optional[dict] = None,
+    ) -> None:
+        self.regressor = regressor
+        self.lags = lags
+        self.transformer_y = transformer_y
+        self.transformer_exog = transformer_exog
+        self.weight_func = weight_func
+        self.differentiation = differentiation
+        self.fit_kwargs = fit_kwargs
+        self.binner_kwargs = binner_kwargs
+
+        super().__init__()
+
+        self._forecaster = None
+        self._transformer_y = None
+        self._transformer_exog = None
+
+        self._clone_estimators()
+
+    def _clone_estimators(self: "SkforecastAutoreg"):
+        """Clone the regressor and transformers."""
+        from sklearn.base import clone
+
+        self._regressor = clone(self.regressor)
+
+        if self.transformer_y:
+            self._transformer_y = clone(self.transformer_y)
+
+        if self.transformer_exog:
+            self._transformer_exog = clone(self.transformer_exog)
+
+    def _create_forecaster(self: "SkforecastAutoreg"):
+        """Create ``skforecast.ForecasterAutoreg.ForecasterAutoreg`` model."""
+        from skforecast.ForecasterAutoreg import ForecasterAutoreg
+
+        return ForecasterAutoreg(
+            self._regressor,
+            self.lags,
+            transformer_y=self._transformer_y,
+            transformer_exog=self._transformer_exog,
+            weight_func=self.weight_func,
+            differentiation=self.differentiation,
+            fit_kwargs=self.fit_kwargs,
+            binner_kwargs=self.binner_kwargs,
+        )
+
+    @staticmethod
+    def _coerce_column_names(X: Optional[pd.DataFrame]):
+        if X is None:
+            return None
+
+        return X.rename(columns=lambda column_name: str(column_name))
+
+    @staticmethod
+    def _coerce_int_to_range_index(df):
+        new_df = df.copy(deep=True)
+        start = new_df.index[0]
+        stop = new_df.index[-1] + 1
+        if len(new_df.index) == 1:
+            step = 1
+        else:
+            step = new_df.index[1] - start
+
+        new_index = pd.RangeIndex(start, stop, step)
+        np.testing.assert_array_equal(new_df.index, new_index)
+        new_df.index = new_index
+        return new_df
+
+    @staticmethod
+    def _coerce_period_to_datetime_index(df):
+        new_df = df.copy(deep=True)
+        period_freq = new_df.index.freq
+        new_df.index = new_df.index.to_timestamp(how="e").normalize()
```

Comment:
It'll be great if you make a comment here explaining that here we are converting period indices to the end date.

Curious, can period be sub-daily? How does this behave then?

Reply:
I just checked, it can be hourly and so on. I've removed the normalize() call so that it works now.